packages/tinybased/src/lib/tinybased.ts

```diff
@@ -128,6 +148,70 @@ export class TinyBased<
     );
   }
 
+  /**
+   * Registers an event handler that will be called any time there is any change to a row in the provided table.
+   * Information about the type of change will be provided to the handler to help simplify writing business logic
+   *
+   * @param table - The name of the table to listen to changes on
+   * @param {(change: RowChange<TBSchema[TTable]>) => void} handler - The handler function that will be called whenever a change occurs to a row in the table
+   * @param [canMutate] - Whether or not the provided handler can make changes back to the TinyBase instance as part of handling the event
+   * @returns - The id of the registered listener which can be used to unsubscribe later using tinybase.store.delListener
+   */
+  public onRowChange<TTable extends OnlyStringKeys<TBSchema>>(
+    table: TTable,
+    handler: (change: RowChange<TBSchema[TTable]>) => void,
+    canMutate = false
+  ) {
+    const listenerId = this.store.addRowListener(
+      table,
+      null,
+      (_store, _table, rowId, getCellChange) => {
+        const operationType: RowChange<TBSchema[TTable]> = (() => {
+          if (!this.store.hasRow(table, rowId)) {
+            return { type: 'delete', rowId };
```

Comment:
Having the types now is pretty sweet, i think this can reduce a bit of the logic i was doing with the files 🥳